client/asset/eth/nodeclient.go

```diff
@@ -0,0 +1,373 @@
+	bn, err := n.ec.BlockNumber(ctx)
+	if err != nil {
+		return nil, err
+	}
+	header, err := n.ec.HeaderByNumber(ctx, big.NewInt(int64(bn)))
+	if err != nil {
+		return nil, err
+	}
+	return header, nil
+}
+
+// block gets the block identified by hash.
+func (n *nodeClient) block(ctx context.Context, hash common.Hash) (*types.Block, error) {
+	block, err := n.ec.BlockByHash(ctx, hash)
+	if err != nil {
+		return nil, err
+	}
+	return block, nil
+}
+
+// accounts returns all accounts from the internal node.
+func (n *nodeClient) accounts() []*accounts.Account {
+	var accts []*accounts.Account
+	for _, wallet := range n.node.AccountManager().Wallets() {
+		for _, acct := range wallet.Accounts() {
+			accts = append(accts, &acct)
+		}
+	}
+	return accts
+}
+
+// balance gets the current balance of an address.
+func (n *nodeClient) balance(ctx context.Context, addr *common.Address) (*big.Int, error) {
+	return n.ec.BalanceAt(ctx, *addr, nil)
+}
+
+// unlock uses a raw request to unlock an account indefinitely.
+func (n *nodeClient) unlock(ctx context.Context, pw string, acct *accounts.Account) error {
+	return n.creds.ks.TimedUnlock(*n.creds.acct, pw, 0)
+}
+
+// lock uses a raw request to unlock an account indefinitely.
+func (n *nodeClient) lock(ctx context.Context, acct *accounts.Account) error {
+	return n.creds.ks.Lock(n.creds.addr)
+}
+
+// transactionReceipt uses a raw request to retrieve a transaction's receipt.
+func (n *nodeClient) transactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error) {
+	return n.ec.TransactionReceipt(ctx, txHash)
+}
+
+// pendingTransactions returns pending transactions.
+func (n *nodeClient) pendingTransactions(ctx context.Context) ([]*types.Transaction, error) {
+	return n.leth.ApiBackend.GetPoolTransactions()
+}
+
+// addPeer adds a peer.
+func (n *nodeClient) addPeer(ctx context.Context, peerURL string) error {
+	peer, err := enode.Parse(enode.ValidSchemes, peerURL)
+	if err != nil {
+		return err
+	}
+	n.p2pSrv.AddPeer(peer)
+	return nil
+}
+
+// blockNumber gets the block number at time of calling.
+func (n *nodeClient) blockNumber(ctx context.Context) (uint64, error) {
+	bn, err := n.ec.BlockNumber(ctx)
+	if err != nil {
+		return 0, err
+	}
+	return bn, nil
+}
+
+// nodeInfo retrieves useful information about a node.
+func (n *nodeClient) nodeInfo(ctx context.Context) *p2p.NodeInfo {
+	return n.p2pSrv.NodeInfo()
+}
+
+// listWallets list all of the wallet's wallets? and accounts along with details
+// such as locked status.
+func (n *nodeClient) listWallets(ctx context.Context) []accounts.Wallet {
+	return n.creds.ks.Wallets()
+}
+
+// sendTransaction uses a raw request to send tx.
+func (n *nodeClient) sendTransaction(ctx context.Context, txOpts *bind.TransactOpts,
+	to common.Address, data []byte) (*types.Transaction, error) {
+
+	nonce, err := n.ec.NonceAt(ctx, n.creds.addr, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error getting nonce: %v", err)
+	}
```

Comment:
It's not impossible that this is insufficient to guarantee a unique nonce. I haven't verified that this is incremented before the transaction is mined.